corehq/apps/toggle_ui/templates/toggle/flags.html

```diff
@@ -58,7 +58,11 @@
       <tbody>
       {% for toggle in toggles %}
         <tr>
-          <td><span class="label label-{{ toggle.tag.css_class }}">{{ toggle.tag.name }}</span></td>
+            <td>
+                {# Put the index here (hidden) so it sorts properly #}
+                <span class="hide">{{ toggle.tag.index }}</span>
```

Comment:
Not proud of this, but I couldn't figure out how to specify an alternative sort or to preserve the ordering used on the initial page render.  Ideally this would also use the name column as a secondary sort, but it doesn't.

Reply:
Not quite following what's going on here. Is the issue that flags get reordered when the user searches or selects a tag group? How does this fix it?

Reply:
No matter what order I send the flags down, after a second or so, the JS kicks in and reorders them by this first column.  This seems to happen alphabetically by the value of the column, so this turns:
```
Deprecated
One-Off / Custom
Product
```
into
```
1 Product
2 One-Off / Custom
3 Deprecated
```
(but with the numbers hidden from the user)

Reply:
Forgot this was datatables...I think you'd want to add a hidden column for the index and then set `order` with a primary and secondary sort, but I don't know exactly what it would look like. Feel free to call that out of scope.

Reply:
It's also old datatables, so instead of `order`, it uses `aaSort` and `bSort` (and possibly others?) - and we have [some sort of wrapper around that](https://github.com/dimagi/commcare-hq/blob/573738a0cca7a229078e6f5b231d312d67ff137a/corehq/apps/reports/static/reports/js/config.dataTables.bootstrap.js#L18-L19) where you pass booleans in for other sort related things.

Reply:
The emoticons available to respond to a comment are sadly limited.
 😨 😞 ought to cover it.